### DISCUSSION TITLE
Clarification on  tableau-scraping customization/extension

### DISCUSSION THREAD

**Discussion by shra-1s** [bertrandmartel/tableau-scraping#65] [Q&A]
@bertrandmartel 

This library looks fantastic. I have done similar (subset) implementation for 'on-prem' Tableau Server with basic authentication flow (local user account having username and password).

Quickly checked tableauscrapper.py but couldn't locate any 'auth' specific code block. Just wondering if this lib can be used on private Tableau server?

Another query I have is: Since we are scrapping data using Tableau VizQL API, what would be the impact of it if we try to automate kind of Email Subscription utility which sends periodic emails with scrapped data in the email attachment?

Complexity: No. of Sites x No. of Subscriptions

Other factors which may impact the performance is: 
- load time of view
- active Tableau users (and kind of interactions) during subscription trigger event occurs
- any background task (e.g. data/extract refresh) running during subscription trigger event occurs

Best,

SS

-- Basic Auth snippet in my current solution
![Basic Auth snippet](https://user-images.githubusercontent.com/110523734/182549407-089cb2f3-c4e9-42ea-bcff-793638bc8563.png)

**Comment by VijrumbhanM** [comment]
@shra-1s 

I have also tried a similar approach for authenticating tableau login before scraping the data. but it didnot work for me? 
vizportal auth is working but iam unable to use the same auth session while scraping the data.
could you pls tell us what should the flow be ? 
presently in my code i am authenticating using vizportal auth after api.setsession() in tableauscraper.py
ex:
def loads(self, url, params={}):
        api.setSession(self)
        api.Auth();  // auth is new function that i have added where i am calling vizportal auth api
        r = api.getTableauViz(self, self.session, url, params)

**Reply by shra-1s** [reply]
@VijrumbhanM 

You can trace request response by loading Tableau dashboard in browser using Fidler/similar tools to know the exact flow.
Moreover the flow is dependent on many considerations. One of them is you want to scrap data 'on behalf of users' or for yourself.

Things to note for 'on behalf of' flow:
- Basic Auth is needed for Server Admin logon
- Post successful login, you need to request Trusted ticket on behalf of user
- Redeem Trusted ticket to load and bootstrap the view

If you just want data to be scraped for your own views:
- Logon with your credential
- Load and bootstrap the view

Best,

SS
